### PR TITLE
[WIP] Update core create interface

### PR DESCRIPTION
**Pull request by asoong**

### CODE REVIEW

File: contracts/core/lib/CoreState.sol [hunk 1]
```diff
@@ -134,35 +128,6 @@ contract CoreState {
         return state.factories;
     }
 
-    /**
-     * Return boolean indicating if address is valid Set.
-     *
-     * @param  _set           Set address
-     * @return bool           Boolean indicating if valid Set
-     */
-    function validSets(
```
Comment (bweick) [inline contracts/core/lib/CoreState.sol]: This needs to be in Core for issue and redeem. How are we supposed to know which factory to query?

File: contracts/core/SetTokenFactory.sol [hunk 2]
```diff
@@ -64,27 +77,93 @@ contract SetTokenFactory
...
+            }
+        }
+    }
+
+    /**
+     * Check mapping for Set validity
+     *
+     * @param  _set   The address of the SetToken to verify
+     */
+    function isSetValid(
```
Comment (felix2feng) [inline contracts/core/SetTokenFactory.sol]: Why are we checking the validity of the Set here? Can't we just check Core?

File: contracts/core/SetTokenFactory.sol [hunk 3]
```diff
@@ -28,32 +27,46 @@ import { SetToken } from "./SetToken.sol";
...
 
     // Address of the Core contract
     address public core;
 
-    /* ============ Setter Functions ============ */
+    // Array of tracked SetTokens
+    address[] public setTokens;
+
+    // Mapping of tracked SetTokens
+    mapping(address => bool) public validSets;
```
Comment (felix2feng) [inline contracts/core/SetTokenFactory.sol]: Don't we do this in Core?

File: contracts/core/lib/CoreState.sol [hunk 4]
```diff
@@ -44,12 +44,6 @@ contract CoreState {
         // Array of tracked SetToken factories
         address[] factories;
 
-        // Mapping of tracked SetTokens
-        mapping(address => bool) validSets;
```
Comment (felix2feng) [inline contracts/core/lib/CoreState.sol]: Why are we moving the tracked Sets into factories?

**Pull request closed without merging**